promql/parser/prettier_rules.md

```diff
@@ -0,0 +1,16 @@
+# Prettifying PromQL expressions
+This files contains rules for prettifying PromQL expressions.
+
+Note: The current version of prettier does not preserve comments
+
+### Keywords
+`max_characters_per_line`: Maximum number of characters that will be allowed on a single line in a prettified PromQL expression
+
+### Rules
+1. A node exceeding the `max_characters_per_line` will qualify for split unless
+   1. It is a `SubqueryExpr` and `MatrixSelector`
```

Comment:
Am I right?
```suggestion
   1. It is a `MatrixSelector`
```

Reply:
Yes, because a subquery contains 2 parts:
1. The internal expression
2. Time suffix

And if a subquery expression will exceed `max_characters_per_line`, then the internal expression will be split the least, meaning, from a user’s perspective, a subquery will qualify for split.